akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -700,6 +706,10 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
 
       debug(s"[$streamId] sending ${toSend.length} bytes, endStream = $endStream, remaining buffer [${buffer.length}], remaining stream-level WINDOW [$outboundWindowLeft]")
 
+      // Multiplexer will enqueue for us if we report more data being available
+      // We cannot call `multiplexer.enqueueOutStream` from here because this is called from the multiplexer.
+      isEnqueued = !isDone && canSend
```

Comment:
This is where all the complexity now went... We need to set a flag here even if the enqueuing happens on the other side in the multiplexer.